Blankscreen-exe/porfolio-website
# Fix Dropdown Menu Closing on Reopen After Outside Click

### Description:
This PR addresses the issue where the dropdown menu closes immediately after clicking the hamburger icon following an outside click.

#### Changes made:
1. Added logic to ensure the `handleClickOutside` event listener does not interfere when the dropdown is reopened.
2. Updated the state management to prevent the dropdown from briefly opening and then closing after an outside click.

#### Steps to Test:
1. Click the hamburger icon to open the dropdown.
2. Click outside to close the dropdown.
3. Click the hamburger icon again – the dropdown should now remain open.

Please review the changes and test the dropdown behavior. Let me know if any further adjustments are needed.
